### ISSUE TITLE
Background jobs (Start-Job) do not respect in-session culture changes

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#13289]
When you make an in-session culture change by changing the current thread's culture by assigning to `[cultureinfo]::CurrentCulture` (the same applies to `[cultureinfo]::CurrentUICulture`), PowerShell commendably propagates this change to:

* new *threads* (`Start-ThreadJob`, `ForEach-Object -Parallel`)

* remotely executed code (e.g., `Invoke-Command -ComputerName. ...`)

Unexpectedly, however, background jobs (`Start-Job`, postpositional `&`) do _not_ propagate such changes.

Note: This discrepancy has always existed, including in Windows PowerShell.

## Steps to reproduce

```powershell
# Note: If 'it-IT' happens to be your current culture, target a different one.
[cultureinfo]::CurrentCulture = 'it-IT'
Start-Job { [cultureinfo]::CurrentCulture.Name } | Receive-Job -Wait -AutoRemoveJob | Should -Be it-IT
```

## Expected behavior

The test should succeed.

## Actual behavior

The test fails, because `Start-Job` executes the code in the child process with the culture that was in effect at session _start-up_ time rather than the currently effective one.

E.g., if the persistently configured culture is `en-US`:

```none
Expected strings to be the same, but they were different. 
String lengths are both 5. Strings differ at index 0. Expected: 'it-IT' But was:  'en-US'
```

## Environment data

```none
PowerShell Core 7.1.0-preview.5
```

**Comment by iSazonov** [comment]
It seems [cultureinfo]::CurrentCulture` is not reliable way to change culture on the fly.

I think you could update #3833 and track the issue there.

**Comment by mklement0** [comment]
#3833 is merely about providing a _PowerShell-idiomatic_ way to change the culture in-session - all it would do behind the scenes is to update `[cultureinfo]::CurrentCulture`.

Updating `[cultureinfo]::CurrentCulture` _is_ the reliable way to change the _current thread_'s culture (WinPS had a bug where _it_ reset the culture after every interactively submitted command, but that no longer applies).

What _this_ issue asks for is that the culture in effect for PowerShell's main thread  be _consistently_ propagated to other threads and child processes:

It already works as expected with PS remoting and `Start-ThreadJob` / `ForEach-Object -Parallel`, but not with `Start-Job`, so `Start-Job` is the place where the fix is required.

**Comment by iSazonov** [comment]
See https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.defaultthreadcurrentculture?view=netcore-3.1

**Comment by mklement0** [comment]
That neither changes the main thread's culture nor does it fix the problem with `Start-Job`.

**Comment by iSazonov** [comment]
This is why this can only be fixed in # 3833

**Comment by mklement0** [comment]
As stated, #3833 would be _syntactic sugar_ for assigning to `[cultureinfo]::CurrentCulture`, and that wouldn't fix the behavior of `Start-Job`.

**Comment by iSazonov** [comment]
Makes no sense to implement Set-Culture as "syntactic sugar for assigning to [cultureinfo]::CurrentCulture".
Only in the cmdlet we can do more work to address all scenarios.

**Comment by mklement0** [comment]
No: Just like PS remoting calls and `Start-ThreadJob` / `ForEach-Object -Parallel` already respect the effective ` [cultureinfo]::CurrentCulture`, so should `Start-Job`, so _it_ needs to be fixed - this is independent of #3833.

**Comment by mklement0** [comment]
And: Making `Set-Culture -Session` syntactic sugar for assigning to `[cultureinfo]::CurrentCulture` _is_ the right implementation, given that the value of `[cultureinfo]::CurrentCulture` is what all culture-aware .NET calls (including cmdlets, scripts, and functions) called from the same thread automatically respect. 

For a shell like PowerShell it makes sense to deliberately extend that behavior to all PowerShell code that is conceived of as executing _in the same session_ - we're mostly there, except for `Start-Job`.
